From dfa04d5311537b2405e3bd251b9b6d334d29e754 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:40 -0400
Subject: [ethernet] igb: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1409062359-13063-23-git-send-email-sassmann@redhat.com>
Patchwork-id: 89141
O-Subject: [RHEL7.1 PATCH 22/81] igb: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range() and pci_enable_msix_range()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Cc: e1000-devel@lists.sourceforge.net
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 479d02dfadfbe850ced61c5c83ca16b8b9d4ac5f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 46d31a4..84dfa3f 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -1111,10 +1111,11 @@ static void igb_set_interrupt_capability(struct igb_adapter *adapter, bool msix)
  for (i = 0; i < numvecs; i++)
   adapter->msix_entries[i].entry = i;
 
- err = pci_enable_msix(adapter->pdev,
-         adapter->msix_entries,
-         numvecs);
- if (err == 0)
+ err = pci_enable_msix_range(adapter->pdev,
+        adapter->msix_entries,
+        numvecs,
+        numvecs);
+ if (err > 0)
   return;
 
  igb_reset_interrupt_capability(adapter);
-- 
1.7.1